processor/src/test/java/org/mapstruct/ap/test/injectionstrategy/spring/annotateWith/qualified/SpringAnnotateWithMapperTest.java

```diff
@@ -0,0 +1,151 @@
+ */
+package org.mapstruct.ap.test.injectionstrategy.spring.annotateWith.qualified;
+
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.extension.RegisterExtension;
+import org.mapstruct.ap.test.injectionstrategy.shared.CustomerDto;
+import org.mapstruct.ap.test.injectionstrategy.shared.CustomerEntity;
+import org.mapstruct.ap.test.injectionstrategy.shared.Gender;
+import org.mapstruct.ap.test.injectionstrategy.shared.GenderDto;
+import org.mapstruct.ap.testutil.IssueKey;
+import org.mapstruct.ap.testutil.ProcessorTest;
+import org.mapstruct.ap.testutil.WithClasses;
+import org.mapstruct.ap.testutil.WithSpring;
+import org.mapstruct.ap.testutil.runner.GeneratedSource;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.context.ConfigurableApplicationContext;
+import org.springframework.context.annotation.AnnotationConfigApplicationContext;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.context.annotation.Configuration;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * Test field injection for component model spring.
+ *
+ * @author Filip Hrisafov
+ */
+@WithClasses({
+        CustomerDto.class,
+        CustomerEntity.class,
+        Gender.class,
+        GenderDto.class,
+        CustomerSpringComponentQualifiedMapper.class,
+        CustomerSpringControllerQualifiedMapper.class,
+        CustomerSpringServiceQualifiedMapper.class,
+        CustomerSpringRepositoryQualifiedMapper.class,
+        CustomStereotype.class,
+        CustomerSpringCustomStereotypeQualifiedMapper.class,
+        CustomerSpringDefaultMapper.class,
+        GenderSpringDefaultMapper.class
+})
+@IssueKey( "1427" )
+@ComponentScan(basePackageClasses = CustomerSpringDefaultMapper.class)
+@Configuration
+@WithSpring
+public class SpringAnnotateWithMapperTest {
+
+    @Autowired
+    @Qualifier( "AnnotateWithComponent" )
+    private CustomerSpringDefaultMapper annotateWithComponentCustomerMapper;
+
+    @Autowired
+    @Qualifier( "AnnotateWithController" )
+    private CustomerSpringDefaultMapper annotateWithControllerCustomerMapper;
+
+    @Autowired
+    @Qualifier( "AnnotateWithService" )
+    private CustomerSpringDefaultMapper annotateWithServiceCustomerMapper;
+
+    @Autowired
+    @Qualifier( "AnnotateWithRepository" )
+    private CustomerSpringDefaultMapper annotateWithRepositoryCustomerMapper;
+
+    @Autowired
+    @Qualifier( "AnnotateWithCustomStereotype" )
+    private CustomerSpringDefaultMapper annotateWithCustomStereotypeCustomerMapper;
+
+    @RegisterExtension
+    final GeneratedSource generatedSource = new GeneratedSource();
+    private ConfigurableApplicationContext context;
+
+    @BeforeEach
+    public void springUp() {
```

Comment:
I don't really think that we need to test this in the spring context. We can just validate the generated sources.

Reply:
Thank you for point that out @filiphr. I updated the code to test only the generated source.